src/lib/database/mysql/temp_user.ts

```diff
@@ -0,0 +1,55 @@
+import type { MittwaldAPIV2 } from "@mittwald/api-client";
+import { assertStatus, MittwaldAPIV2Client } from "@mittwald/api-client";
+import { randomBytes } from "crypto";
+import assertSuccess from "../../assert_success.js";
+
+type DatabaseMySqlUser = MittwaldAPIV2.Components.Schemas.DatabaseMySqlUser;
+
+export interface TemporaryUser {
+  user: DatabaseMySqlUser;
+  password: string;
+
+  cleanup(): Promise<void>;
+}
+
+/**
+ * Creates a temporary user for a database operation.
+ *
+ * Caution: The returned TemporaryUser object contains a "cleanup()" functions;
+ * callers of the createTemporaryUser function must make sure to call this
```

Comment:
> „Functions should not have side effects“ \
> _https://www.youtube.com/watch?v=7EmboKQH8lM&t=5242s_

Reply:
Thanks for the Uncle Bob Bomb... 😉  Refactored in d508120.